Engine/DataFeeds/LiveOptionChainProvider.cs

```diff
@@ -42,18 +62,158 @@ static LiveOptionChainProvider()
+                        .ReadAsStringAsync()
+                        .SynchronouslyAwaitTaskResult());
+
+                    // For now, only support American options on CME
+                    var selectedOption = tradesAndExpiriesResponse
+                        .FirstOrDefault(x => !x.Daily && !x.Weekly && !x.Sto && x.OptionType == "AME");
+
+                    if (selectedOption == null)
+                    {
+                        Log.Error($"LiveOptionChainProvider.GetFutureOptionContractList(): Found no matching future options for contract {futureContractSymbol}");
+                        yield break;
+                    }
+
+                    // Gather the month code and the year's last number to query the next API, which expects an expiration as `<MONTH_CODE><YEAR_LAST_NUMBER>`
+                    var futureContractExpiration = selectedOption.Expirations
+                        .FirstOrDefault(x => x.Expiration.Year == futureContractSymbol.ID.Date.Year && x.Expiration.Month == futureContractSymbol.ID.Date.Month);
+
+                    if (futureContractExpiration == null)
+                    {
+                        Log.Error($"LiveOptionChainProvider.GetFutureOptionContractList(): Found no future options with matching expiry year and month for contract {futureContractSymbol}");
+                        yield break;
+                    }
+
+                    var futureContractMonthCode = futureContractExpiration.Expiration.Code;
+                    var futureContractId = futureContractExpiration.ContractId;
+
+                    _rateGate.WaitToProceed();
+
+                    var mhdbEntry = MarketHoursDatabase.FromDataFolder()
+                        .GetExchangeHours(futureContractSymbol.ID.Market, futureContractSymbol, futureContractSymbol.SecurityType);
+                    // We want at least yesterday's trading day, not today since settlement hasn't occurred yet.
+                    var lastTradeDate = mhdbEntry.GetPreviousTradingDay(DateTime.UtcNow.AddDays(-1).ConvertFromUtc(mhdbEntry.TimeZone));
+
+                    // Subtract one day from now for settlement API since settlement may not be available for today yet
+                    var optionChainQuotesResponseResult = _client.GetAsync(CMEOptionChainURL.Replace(CMEProductCodeReplace, selectedOption.ProductId.ToStringInvariant())
+                        .Replace(CMEContractCodeReplace, futureContractId)
+                        .Replace(CMEProductExpirationReplace, futureContractMonthCode)
+                        .Replace(CMEDateTimeReplace, lastTradeDate.ToStringInvariant("MM/dd/yyyy"))
+                        + Math.Floor((DateTime.UtcNow - _epoch).TotalMilliseconds).ToStringInvariant());
+
+                    optionChainQuotesResponseResult.Result.EnsureSuccessStatusCode();
+
+                    var futureOptionChain = JsonConvert.DeserializeObject<CMEOptionChainSettlements>(optionChainQuotesResponseResult.Result.Content
```

Comment:
For the unit test I'm getting `{"settlements":[],"updateTime":"-","dsHeader":"E-mini S&P 500  American Options","reportType":"Final","tradeDate":"-","empty":true}` for `optionChainQuotesResponseResult.Result.Content`